lib/trino-parquet/src/main/java/io/trino/parquet/BloomFilterStore.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.airlift.log.Logger;
+import org.apache.parquet.column.values.bloomfilter.BlockSplitBloomFilter;
+import org.apache.parquet.column.values.bloomfilter.BloomFilter;
+import org.apache.parquet.format.BloomFilterHeader;
+import org.apache.parquet.format.Util;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnPath;
+import org.apache.parquet.io.SeekableInputStream;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class BloomFilterStore
+{
+    private static final Logger LOG = Logger.get(BloomFilterStore.class);
+
+    private final Map<ColumnPath, BloomFilter> cache = new HashMap<>();
+    private final ParquetDataSource dataSource;
+
+    @VisibleForTesting
+    BloomFilterStore()
+    {
+        this.dataSource = null;
+    }
+
+    public BloomFilterStore(ParquetDataSource dataSource)
+    {
+        this.dataSource = dataSource;
+    }
+
+    @VisibleForTesting
+    void insertBloomFilter(ColumnPath columnPath, BloomFilter bloomFilter)
+    {
+        cache.put(columnPath, bloomFilter);
+    }
+
+    public Optional<BloomFilter> readBloomFilter(ColumnChunkMetaData columnMetaData)
+    {
+        if (!bloomFilterEnabled(columnMetaData)) {
+            return Optional.empty();
+        }
+
+        if (cache.containsKey(columnMetaData.getPath())) {
+            return Optional.ofNullable(cache.get(columnMetaData.getPath()));
+        }
+
+        try {
+            Optional<BloomFilter> bloomFilterOptional = readBloomFilterInternal(columnMetaData);
+            if (bloomFilterOptional.isEmpty()) {
+                cache.put(columnMetaData.getPath(), null);
+            }
+            else {
+                cache.put(columnMetaData.getPath(), bloomFilterOptional.get());
+            }
+            return bloomFilterOptional;
+        }
+        catch (IOException exception) {
+            LOG.error(exception, "Failed to read Bloom filter data");
```

Comment:
`throw new RuntimeException(e);`
These failures should not be hidden

Reply:
done.

Reply:
only catch exception when deserializing header, since we don't need to deserialize the data part. Just read in as a byte array.